File: izmk-common/src/main/java/ovo/xsvf/Pair.java
package ovo.xsvf;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@ToString
@EqualsAndHashCode
@AllArgsConstructor
public class Pair<T, U> {
    public T first;
    public U second;

    public static <T, U> Pair<T, U> of(T first, U second) {
        return new Pair<>(first, second);
    }
}

File: izmk-common/src/main/kotlin/ovo/xsvf/Pair.kt
package ovo.xsvf

/**
 * @author LangYa466
 * @since 2025/2/17
 */
data class Pair<T, U>(
    @JvmField var first: T,
    @JvmField var second: U
) {
    companion object {
        @JvmStatic
        fun <T, U> of(first: T, second: U): Pair<T, U> = Pair(first, second)
    }
}
